### PR TITLE
Add an async version of transactions

### LINKED ISSUE #43: JS async transactions
Transactions are now supported in the sync version, an async version still needs to be done. I'm guessing it will need to be promise-based, which doesn't quite fit with our callback async calls.

Option a: dump the responsibility of promisifying the responsibilities to the user, and just rollback the transaction on reject.
Option b: migrate or create yet another version of the async callbacks to return promises.

### PR DESCRIPTION
**Pull request by ospfranco**
closes #43 

@EduFrazao can you test if it is working properly on your end?

### CODE REVIEW

**Comment by ospfranco** [comment]
@EduFrazao did you have some time to test this?

**Comment by EduFrazao** [comment]
> @EduFrazao did you have some time to test this?

Hi @ospfranco . I'm really sorry but one of applications (that makes heavy use of quick-sqlite) has been deployed in production this week, replacing an old Cordova/IONIC. This has been a busy week. I wil do my best to test this changes today and let you know. Sorry for this delay.

File: src/index.ts [hunk 1]
```diff
@@ -156,6 +165,10 @@ interface ISQLite {
   ) => StatementResult;
   detach: (mainDbName: string, alias: string) => StatementResult;
   transaction: (dbName: string, fn: (tx: Transaction) => boolean) => void;
+  asyncTransaction: (
```
Comment (EduFrazao) [inline src/index.ts:168]: Could you add the asyncTransaction method to IDBConnection to keep it feature equivalent with ISQLite?

**Review by EduFrazao — COMMENTED** [review]
Hi @ospfranco. Sorry for the late response. I've made many tests with a bunch of large transactions running on the Async API and everything seems fine.
I only added asyncTransaction to IDBConnection because I like to use it on my projects, as I use a wrapper around a opened connections that holds a reference to it.

**Pull request merged**